src/bandersnatch/package.py

```diff
@@ -115,13 +110,15 @@ async def fetch_metadata(self, attempts: int = 3) -> Any:
                 )
                 raise StaleMetadata(package_name=self.name, attempts=attempts)
 
-    async def sync(self, attempts: int = 3) -> None:
+    async def sync(self, filters: "LoadedFilters", attempts: int = 3) -> None:
```

Comment:
If we have other external people using this API, this is a pretty big change.

Reply:
I agree.  I'm trying to coordinate my changes with Daniel's PR and since we want to remove the Mirror reference from Package but still allow Package to do its own filtering this is the solution we settled on.  Any ideas for how to improve this would be helpful.